source/developer/mobile-developer-setup.rst

```diff
@@ -117,10 +188,128 @@ Android (Device)
+
+1. Fork the `mattermost-mobile <https://github.com/mattermost/mattermost-mobile>`_ repository on GitHub.
+2. Clone your fork locally:
+ - Open a terminal 
+ - Change to a directory you want to hold your local copy 
+ - Run ``git clone https://github.com/<username>/mattermost-mobile.git`` if you want to use HTTPS, or ``git clone git@github.com:<username>/mattermost-mobile.git`` if you want to use SSH
+
+.. important::
+  ``<username>`` refers to the username or organization in GitHub that forked the repository.
+
+3. Change the directory to ``mattermost-mobile``.
+
+  ``cd mattermost-mobile``
+
+4. Run ``make pre-run`` in order to install all the dependencies.
+
+
+.. important::
+  It is important that you run everything with the make commands and avoid using npm or yarn to install dependencies. If you use npm or yarn, you may skip steps and the app won't build correctly.
+ 
+Make Commands Explained
+------------------------------------
+
+We've included several make commands to control the development flow and to ensure that everything works as expected. Always use these make commands unless what you're trying to do can't be accomplished by them.
+
+Every make command has to be run from a terminal in the project's root directory.
+
+Commands to Prepare the App
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+These make commands are used to install dependencies, to configure necessary steps before running or building the app, and to clean everything.
+
+ - **make pre-run**: Downloads any project dependencies needed and sets up the app assets. This is the equivalent of running  ``.yarninstall``, ``.podinstall`` and ``dist/assets`` make commands.
+ - **make .yarninstall**: Downloads JavaScript and react native dependencies. Once this command finishes executing it will create a ``.yarninstall`` empty file in the project's root directory to indicate that this command already ran. If yarn updates your local dependencies in the ``node_modules`` directory, it will automatically run ``make post-install`` for you.
+ - **make .podinstall**: Downloads cocoapods dependencies needed to build the iOS app. Once this command finishes executing it will create a ``.podinstall`` empty file in the project's root directory to indicate that this command already ran.
+ - **make post-install**: Normally this command runs automatically if yarn detects that the project updated some or all of its JavaScript dependencies. Once yarn finishes executing, this command automatically sets up a few post installation steps to ensure that everything runs correctly.
+ - **make clean**: Removes all downloaded dependencies, clears the cache of those dependencies and deletes any builds that were created. It will not reset the repo, so your current changes will still be there.
+ - **make dist/assets**: Builds the assets to be used including images, localization files and overriding of asset files (see Override Assets & White Labeling for details).
```

Comment:
Do we have a link for `Override Assets & White Labeling`? Is it in our docs somewhere?

Reply:
that is not yet documented